### PR TITLE
Refactor 'count_uses_of_property' method in crop model

### PR DESCRIPTION
**Pull request by alran**
Solves issue [1087](https://github.com/Growstuff/growstuff/issues/1087), which follows onto [1086](https://github.com/Growstuff/growstuff/pull/1086)

### CODE REVIEW

**Comment by CloCkWeRX** [comment]
Should that be something more like SELECT COUNT(colname) FROM ... WHERE
colname IS NOT NULL AND colname <> '';
?


On 24 Dec 2016 4:11 AM, "Alyssa Ransbury" <notifications@github.com> wrote:

Solves issue 1087 <https://github.com/Growstuff/growstuff/issues/1087>,
which follows onto 1086 <https://github.com/Growstuff/growstuff/pull/1086>
------------------------------
You can view, comment on, or merge this pull request online at:

  https://github.com/Growstuff/growstuff/pull/1135
Commit Summary

   - Refactor 'count_uses_of_property' method in crop model

File Changes

   - *M* app/models/crop.rb
   <https://github.com/Growstuff/growstuff/pull/1135/files#diff-0> (8)

Patch Links:

   - https://github.com/Growstuff/growstuff/pull/1135.patch
   - https://github.com/Growstuff/growstuff/pull/1135.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/Growstuff/growstuff/pull/1135>, or mute the thread

**Comment by alran** [comment]
@CloCkWeRX you are right - I updated the query to not include nil column values

**Comment by pozorvlak** [comment]
@alran: thanks for the PR! I haven't had time to review it properly yet, but please add yourself to the file CONTRIBUTORS.md. We really need to find a friendlier way of indicating that than a test-suite failure...

File: app/models/crop.rb [hunk 1]
```diff
@@ -350,11 +350,7 @@ def name_already_exists(list_name, name)
   end
 
   def count_uses_of_property(col_name)
-    data = Hash.new(0)
-    plantings.each do |p|
-      data[p.send("#{col_name}")] += 1 if !p.send("#{col_name}").blank?
-    end
-    data
+    plantings.unscoped.where(crop_id: id).where.not("#{col_name}": nil).group(col_name).count
```
Comment (pozorvlak) [inline app/models/crop.rb]: Can we remove the `#{ }` around `col_name`? It should be a no-op, AFAICT.

Comment (pozorvlak) [inline app/models/crop.rb]: Otherwise, looks great!

**Comment by pozorvlak** [comment]
Would anyone (especially @alran) mind if I re-opened https://github.com/alran/growstuff/pull/1 here, possibly with the string-interpolation nitpick fixed? I don't want to get @alran's name wrong in CONTRIBUTORS.md, but the one I used there _is_ plastered all over her GitHub...

File: app/models/crop.rb (hunk 1 above)
Comment (pozorvlak) [inline app/models/crop.rb]: Aaah, if you delete the #{ }, it becomes `col_name: nil` and `col_name` is interpreted as a symbol, not the name of a variable. I've replaced the colon with `=>`.

**Comment by CloCkWeRX** [comment]
Superceded by  #1183 - thanks for starting this one @alran  :)

**Pull request closed without merging**